sonar-scanner-engine/src/main/java/org/sonar/scanner/issue/ignore/scanner/IssueExclusionsRegexpScanner.java

```diff
@@ -116,14 +83,33 @@ public void scan(String resource, Path filePath, Charset encoding) throws IOExce
     if (!lineExclusions.isEmpty()) {
       Set<LineRange> lineRanges = convertLineExclusionsToLineRanges();
       LOG.debug("- Line exclusions found: {}", lineRanges);
-      exclusionPatternInitializer.getPatternMatcher().addPatternToExcludeLines(resource, lineRanges);
+      patternMatcher.addPatternToExcludeLines(componentKey, lineRanges);
     }
   }
 
+  private void processLine(String line) {
+    if (line.trim().length() == 0) {
+      return;
+    }
+
+    // first check the single regexp patterns that can be used to totally exclude a file
+    for (java.util.regex.Pattern pattern : allFilePatterns) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this fully qualified name with "Pattern" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)